packages/hdwallet-portis/src/bitcoin.ts

```diff
@@ -0,0 +1,280 @@
+    return unknown
+
+  if ((path[0] & 0x80000000) >>> 0 !== 0x80000000)
+    return unknown
+
+  let purpose = path[0] & 0x7fffffff
+
+  if (![44, 49, 84].includes(purpose))
+    return unknown
+
+  if (purpose === 44 && scriptType !== BTCInputScriptType.SpendAddress)
+    return unknown
+
+  if (purpose === 49 && scriptType !== BTCInputScriptType.SpendP2SHWitness)
+    return unknown
+
+  if (purpose === 84 && scriptType !== BTCInputScriptType.SpendWitness)
+    return unknown
+
+  let wholeAccount = path.length === 3
+
+  let script = {
+    [BTCInputScriptType.SpendAddress]: ['Legacy'],
+    [BTCInputScriptType.SpendP2SHWitness]: [],
+    [BTCInputScriptType.SpendWitness]: ['Segwit Native']
+  }[scriptType]
+
+  let isPrefork = false
+  if (path[1] !== 0x80000000 + slip44ByCoin(coin)) {
+    switch (coin) {
+    case 'BitcoinCash':
+    case 'BitcoinGold': {
+      if (path[1] === 0x80000000 + slip44ByCoin('Bitcoin')) {
+        isPrefork = true
+        break
+      }
+      return unknown
+    }
+    case 'BitcoinSV': {
+      if (path[1] === 0x80000000 + slip44ByCoin('Bitcoin') ||
+          path[1] === 0x80000000 + slip44ByCoin('BitcoinCash')) {
+        isPrefork = true
+        break
+      }
+      return unknown
+    }
+    default:
+      return unknown
+    }
+  }
+
+  let attributes = isPrefork ? ['Prefork'] : []
+  switch (coin) {
+  case 'Bitcoin':
+  case 'Litecoin':
+  case 'BitcoinGold':
+  case 'Testnet': {
+    attributes = attributes.concat(script)
+    break
+  }
+  default:
+    break
+  }
+
+  let attr = attributes.length ? ` (${attributes.join(', ')})` : ''
+
+  let accountIdx = path[2] & 0x7fffffff
+
+  if (wholeAccount) {
+    return {
+      coin,
+      verbose: `${coin} Account #${accountIdx}${attr}`,
+      accountIdx,
+      wholeAccount: true,
+      isKnown: true,
+      scriptType,
+      isPrefork
+    }
+  } else {
+    let change = path[3] === 1 ? 'Change ' : ''
+    let addressIdx = path[4]
+    return {
+      coin,
+      verbose: `${coin} Account #${accountIdx}, ${change}Address #${addressIdx}${attr}`,
+      accountIdx,
+      addressIdx,
+      wholeAccount: false,
+      isKnown: true,
+      isChange: path[3] === 1,
+      scriptType,
+      isPrefork
+    }
+  }
+}
+
+
+export async function btcGetAddress (msg: BTCGetAddress, portis: any): Promise<string> {
+
+  if(!msg.addressNList.length)
+    throw new Error('Empty addressNList')
+
+  const scriptType = msg.scriptType
+  const purpose = msg.addressNList[0]
+
+  const hardPath = hardenedPath(msg.addressNList)
+  const hardPathString = 'm/'+hardPath.reduce( (acc, val, i) => (i !== hardPath.length - 1) ? acc + (val - 2147483648) + `'/` : acc + (val - 2147483648) + `'`, '')
+
+  const { result: xpub } = await portis.getExtendedPublicKey(hardPathString, "Bitcoin")
+
+  const relPath = relativePath(msg.addressNList)
+  const relPathString = relPath.reduce( (acc, val, i) => (i !== relPath.length - 1) ? acc + val + '/' : acc + val, '')
```

Comment:
```suggestion
  const relPathString = addressNListToBip32(relPath)
```

Reply:
and actually, you don't even need it as a string anyway... just pass relPath where you're passing relPathString

Reply:
doing addressNListToBip32(relPath) here returns the string `m/0/0` instead of `0/0`

passing `m/0/0` to derivePath causes the error:

```
ncaught (in promise) TypeError: Expected master, got child
    at BIP32.derivePath (bip32.js:168)
    at Object.<anonymous> (bitcoin.ts:153)
    at step (index.js:26)
    at Object.next (index.js:26)
    at fulfilled (index.js:26)
```

Reply:
Also cant pass relPath instead of relPathString into derivePath() because it expects a string not an array

Reply:
hm. thought it supported that

Reply:
maybe `addressNListToBip32(relPath).substr(2, -1)`?

Reply:
ya thats probably easier to read

Reply:
I think we want:

 const relPathString = addressNListToBIP32(relPath).substr(2)